### PR TITLE
PLAT-136770: Update ImageItem

### PR DESCRIPTION
**Pull request by paul-beldean-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
ImageItem component needs to be updated to the latest design changes

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Added styling for ImageItem based on the latest UX design for Silicon

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-136770

### Comments

Enact-DCO-1.0-Signed-off-by: Paul Beldean paul.beldean@lgepartner.com

### CODE REVIEW

**Review by daniel-stoian-lgp — CHANGES_REQUESTED** [review]
Silicon styling issues:
1. Padding right (30px) for horizontal imageItem is missing
2. Horizontal ImageItem Caption height should be 81px, not 60px - for sizing Fill and None

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ The following is a curated list of changes in the Enact agate module, newest cha
 
 - `agate/Button` to apply active styling only when it is not disabled
 - `agate/ImageItem` width/height proportion when `orientation` is `horizontal`
+- `agate/ImageItem` style to match latest design for Silicon skin
```
Comment (jeonghee27) [inline CHANGELOG.md:25]: Could you please write another changelog about sizing of imageItem on "Added"?

**Review by jeonghee27 — COMMENTED** [review]
Could you add Image.module.less for Image sampler?

.image {
	width: 400px;
	height: 300px;
}

Because  there is a problem if App developer assign the width and height as a style. so we need an example that shows using the public class ".image".

**Comment by jeonghee27** [comment]
Sizing does not work for vertical ImageItems.
GUI  defines this vaguely, and I am not sure if this is necessary. so I think we can put it later if necessary.
Please judge when merge. (mikyung or Seungcheon)

**Comment by paul-beldean-lgp** [comment]
> Sizing does not work for vertical ImageItems.
> GUI defines this vaguely, and I am not sure if this is necessary. so I think we can put it later if necessary.
> Please judge when merge. (mikyung or Seungcheon)

Yes, I implemented it only for horizontal ImageItem like it was in the design.

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: ImageItem/ImageItem.module.less [hunk 2]
```diff
@@ -21,18 +21,46 @@
...
+
+			&.fill {
+				.image {
+					height: @agate-imageitem-horizontal-image-sizing-fill-height;
+				}
+			}
+
+			&.none {
+				.image {
+					.margin-start-end(@agate-imageitem-horizontal-image-sizing-fit-margin);
```
Comment (0x64) [inline ImageItem/ImageItem.module.less:40]: It looks that `fit` and `none` styles are mistaken.

File: ImageItem/ImageItem.js [hunk 3]
```diff
@@ -113,16 +131,16 @@ const ImageItemBase = kind({
...
 	},
 
-	render: ({captionPosition, children, css, disabled, orientation, src, ...rest}) => {
+	render: ({captionPosition, children, css, disabled, orientation, skin, src, ...rest}) => {
 		const [Component, marqueeProps] = (children && (orientation === 'horizontal' || captionPosition === 'below')) ? [MarqueeImageItem, {
-			alignment: 'center'
+			alignment: skin === 'silicon' && orientation === 'horizontal' ? 'left' : 'center'
 		}] : [UiImageItem, null];
 
+		delete rest.sizing;
```
Comment (0x64) [inline ImageItem/ImageItem.js:142]: Is there any reason to not pass `sizing` prop to 'Component'? Internally, 'Image' component is used for an image and it already supports `sizing` prop via `ui/Image` component. We can set margin for 'silicon' skin, but it looks that `sizing` needs to be handled by `ui/Image` in general.

**Comment by 0x64** [comment]
Note: I rebased the PR based on develop branch with React 17 migration.

File: ImageItem/ImageItem.js (hunk 3 above)
Comment (paul-beldean-lgp) [reply ImageItem/ImageItem.js]: I reverted the change, the initial motive was to not affect the other skins as they did not have sizing specifications.

**Pull request merged**